### ISSUE TITLE
`serverpod create app_name` breaks with `ProcessException: The system cannot find the file specified.` error in windows

### ISSUE THREAD

**Issue by yahu1031** [serverpod/serverpod#106]
## What I have done ##
Installed serverpod_cli in windows. Tried creating `myapp`. But before that, I tried deleting the existing app. When I create an app, It is created but it fails in the last step - `flutter create .`. These are logs.

**Deleted existing app**
```console
PS C:\Users\yaswa> rm .\myapp\

Confirm
The item at C:\Users\yaswa\myapp\ has children and the Recurse parameter was not specified. If you continue, all
children will be removed with the item. Are you sure you want to continue?
[Y] Yes  [A] Yes to All  [N] No  [L] No to All  [S] Suspend  [?] Help (default is "Y"): A
PS C:\Users\yaswa> ls my*
PS C:\Users\yaswa>
```
**Created a new app**
```console
PS C:\Users\yaswa> serverpod create myapp
WARNING! Windows is not officially supported yet. Things may or may not work as expected.

Creating project myapp...
  C:\Users\yaswa/myapp/myapp_serveranalysis_options.yaml
  C:\Users\yaswa/myapp/myapp_serverbin/main.dart
  C:\Users\yaswa/myapp/myapp_serverCHANGELOG.md
  C:\Users\yaswa/myapp/myapp_serverconfig/development.yaml
  C:\Users\yaswa/myapp/myapp_serverconfig/generator.yaml
  C:\Users\yaswa/myapp/myapp_serverconfig/passwords.yaml
  C:\Users\yaswa/myapp/myapp_serverconfig/production.yaml
  C:\Users\yaswa/myapp/myapp_serverconfig/staging.yaml
  C:\Users\yaswa/myapp/myapp_serverdocker-compose.yaml
  C:\Users\yaswa/myapp/myapp_servergenerated/protocol.yaml
  C:\Users\yaswa/myapp/myapp_servergenerated/tables-serverpod.pgsql
  C:\Users\yaswa/myapp/myapp_servergenerated/tables.pgsql
  C:\Users\yaswa/myapp/myapp_servergitignore
  C:\Users\yaswa/myapp/myapp_serverlib/server.dart
  C:\Users\yaswa/myapp/myapp_serverlib/src/endpoints/example_endpoint.dart
  C:\Users\yaswa/myapp/myapp_serverlib/src/future_calls/example_future_call.dart
  C:\Users\yaswa/myapp/myapp_serverlib/src/generated/endpoints.dart
  C:\Users\yaswa/myapp/myapp_serverlib/src/generated/example_class.dart
  C:\Users\yaswa/myapp/myapp_serverlib/src/generated/protocol.dart
  C:\Users\yaswa/myapp/myapp_serverlib/src/protocol/example_class.yaml
  C:\Users\yaswa/myapp/myapp_serverpubspec.yaml
  C:\Users\yaswa/myapp/myapp_serverREADME.md
  C:\Users\yaswa/myapp/myapp_serversetup-tables
  C:\Users\yaswa/myapp/myapp_clientanalysis_options.yaml
  C:\Users\yaswa/myapp/myapp_clientCHANGELOG.md
  C:\Users\yaswa/myapp/myapp_clientgitignore
  C:\Users\yaswa/myapp/myapp_clientlib/PROJECTNAME_client.dart
  C:\Users\yaswa/myapp/myapp_clientlib/src/protocol/client.dart
  C:\Users\yaswa/myapp/myapp_clientlib/src/protocol/example_class.dart
  C:\Users\yaswa/myapp/myapp_clientlib/src/protocol/protocol.dart
  C:\Users\yaswa/myapp/myapp_clientpubspec.yaml
  C:\Users\yaswa/myapp/myapp_clientREADME.md
  C:\Users\yaswa/myapp/myapp_flutteranalysis_options.yaml
  C:\Users\yaswa/myapp/myapp_fluttergitignore
  C:\Users\yaswa/myapp/myapp_flutterlib/main.dart
  C:\Users\yaswa/myapp/myapp_flutterpubspec.yaml
  C:\Users\yaswa/myapp/myapp_flutterREADME.md
  C:\Users\yaswa/myapp/myapp_fluttertest/widget_test.dart
Running `dart pub get` in C:\Users\yaswa/myapp/myapp_server
Resolving dependencies...
+ args 2.3.0
+ async 2.8.2
+ buffer 1.1.1
+ charcode 1.3.1
+ collection 1.16.0
+ crypto 3.0.1
+ executor 2.2.2
+ http 0.13.4
+ http_parser 4.0.0
+ lints 1.0.1
+ meta 1.7.0
+ path 1.8.1
+ pedantic 1.11.1
+ postgres 2.4.3
+ redis 2.2.0 (3.0.0 available)
+ retry 3.1.0
+ sasl_scram 0.1.0
+ saslprep 1.0.2
+ serverpod 0.9.5
+ serverpod_client 0.9.5
+ serverpod_postgres_pool 2.1.1
+ serverpod_serialization 0.9.5
+ serverpod_service_client 0.9.5
+ serverpod_shared 0.9.5
+ source_span 1.8.2
+ stack_trace 1.10.0
+ stream_channel 2.1.0
+ string_scanner 1.1.0
+ term_glyph 1.2.0
+ typed_data 1.3.0
+ unorm_dart 0.2.0
+ vm_service 8.2.1
+ web_socket_channel 2.1.0
+ yaml 3.1.0
Changed 34 dependencies!

Running `dart pub get` in C:\Users\yaswa/myapp/myapp_client
Resolving dependencies...
+ async 2.8.2
+ charcode 1.3.1
+ collection 1.16.0
+ crypto 3.0.1
+ http 0.13.4
+ http_parser 4.0.0
+ meta 1.7.0
+ path 1.8.1
+ serverpod_client 0.9.5
+ serverpod_serialization 0.9.5
+ source_span 1.8.2
+ stream_channel 2.1.0
+ string_scanner 1.1.0
+ term_glyph 1.2.0
+ typed_data 1.3.0
+ web_socket_channel 2.1.0
+ yaml 3.1.0
Changed 17 dependencies!

Running `flutter create` in C:\Users\yaswa/myapp/myapp_flutter
Unhandled exception:
ProcessException: The system cannot find the file specified.

  Command: flutter create .
#0      _ProcessImpl._start (dart:io-patch/process_patch.dart:401:33)
#1      Process.start (dart:io-patch/process_patch.dart:38:20)
#2      _runNonInteractiveProcess (dart:io-patch/process_patch.dart:578:18)
#3      Process.run (dart:io-patch/process_patch.dart:49:12)
#4      CommandLineTools.flutterCreate (file:///C:/Users/yaswa/AppData/Local/Pub/Cache/hosted/pub.dartlang.org/serverpod_cli-0.9.5/bin/create/command_line_tools.dart:16:32)
#5      performCreate (file:///C:/Users/yaswa/AppData/Local/Pub/Cache/hosted/pub.dartlang.org/serverpod_cli-0.9.5/bin/create/create.dart:210:28)
<asynchronous suspension>
#6      main (file:///C:/Users/yaswa/AppData/Local/Pub/Cache/hosted/pub.dartlang.org/serverpod_cli-0.9.5/bin/serverpod_cli.dart:218:9)
<asynchronous suspension>
WARNING! Windows is not officially supported yet. Things may or may not work as expected.

Project myapp already exists.
```

**Comment by yahu1031** [comment]
I know a fix for this. I working on it.

**Comment by saiful00000** [comment]
Did you found any fix?

**Comment by vlidholt** [comment]
There is a PR here: https://github.com/serverpod/serverpod/pull/94 I haven't had a chance to try it out just yet. Need to find a way to run Windows.

**Comment by yahu1031** [comment]
> Did you found any fix?

I tried but there is some issue with the `flutter` command with the project directory.

@vlidholt Instead using `Platform.pathSeparator`, we can use `join(path, path1)` by `path` package. That will avoid most of the path issues.

**Comment by yahu1031** [comment]
I have passed the directory like this `flutter create $dir_path` and remove `workingDir` parameter but still issue with flutter.

**Comment by yahu1031** [comment]
I can make a PR to jump over the #94 but still need some files to be changed. I will do it and make a PR @vlidholt

**Comment by vlidholt** [comment]
Thank you @yahu1031

**Comment by anandsubbu007** [comment]
hi, may i know how long it will take serverpod to work on Windows?

**Comment by yahu1031** [comment]
@anandsubbu007 Not sure, currently as of this issue it needs a fix for dynamic paths. But other than that, They need to work on a lot of other things like scripts. EG: sh script for Mac and Linux, they need write a PowerShell or batch scripts for windows to work.

**Comment by vlidholt** [comment]
I've added a bunch of changes to support Window, but couldn't test it in the virtual machine on my Mac as Docker didn't run in it. If someone want to give 0.9.6 and see if it works that would be awesome!

Otherwise, I'll have to go get a Windows machine.

**Comment by yahu1031** [comment]
It still fails @vlidholt 
```console
> serverpod create my_appsss
WARNING! Windows is not officially supported yet. Things may or may not work as expected.

WARNING! skipping check to see if "dart" exists.
WARNING! skipping check to see if "flutter" exists.
Downloading templates for version 0.9.6
Download complete.

WARNING! skipping check to see if "docker" exists.
Creating project my_appsss...
  C:\Users\yaswa\my_appsss\my_appsss_serveranalysis_options.yaml
  C:\Users\yaswa\my_appsss\my_appsss_serverbin/main.dart
  C:\Users\yaswa\my_appsss\my_appsss_serverCHANGELOG.md
  C:\Users\yaswa\my_appsss\my_appsss_serverconfig/development.yaml
  C:\Users\yaswa\my_appsss\my_appsss_serverconfig/generator.yaml
  C:\Users\yaswa\my_appsss\my_appsss_serverconfig/passwords.yaml
  C:\Users\yaswa\my_appsss\my_appsss_serverconfig/production.yaml
  C:\Users\yaswa\my_appsss\my_appsss_serverconfig/staging.yaml
  C:\Users\yaswa\my_appsss\my_appsss_serverdocker-compose.yaml
  C:\Users\yaswa\my_appsss\my_appsss_servergenerated/protocol.yaml
  C:\Users\yaswa\my_appsss\my_appsss_servergenerated/tables-serverpod.pgsql
  C:\Users\yaswa\my_appsss\my_appsss_servergenerated/tables.pgsql
  C:\Users\yaswa\my_appsss\my_appsss_servergitignore
  C:\Users\yaswa\my_appsss\my_appsss_serverlib/server.dart
  C:\Users\yaswa\my_appsss\my_appsss_serverlib/src/endpoints/example_endpoint.dart
  C:\Users\yaswa\my_appsss\my_appsss_serverlib/src/future_calls/example_future_call.dart
  C:\Users\yaswa\my_appsss\my_appsss_serverlib/src/generated/endpoints.dart
  C:\Users\yaswa\my_appsss\my_appsss_serverlib/src/generated/example_class.dart
  C:\Users\yaswa\my_appsss\my_appsss_serverlib/src/generated/protocol.dart
  C:\Users\yaswa\my_appsss\my_appsss_serverlib/src/protocol/example_class.yaml
  C:\Users\yaswa\my_appsss\my_appsss_serverpubspec.yaml
  C:\Users\yaswa\my_appsss\my_appsss_serverREADME.md
  C:\Users\yaswa\my_appsss\my_appsss_serversetup-tables
  C:\Users\yaswa\my_appsss\my_appsss_clientanalysis_options.yaml
  C:\Users\yaswa\my_appsss\my_appsss_clientCHANGELOG.md
  C:\Users\yaswa\my_appsss\my_appsss_clientgitignore
  C:\Users\yaswa\my_appsss\my_appsss_clientlib/PROJECTNAME_client.dart
  C:\Users\yaswa\my_appsss\my_appsss_clientlib/src/protocol/client.dart
  C:\Users\yaswa\my_appsss\my_appsss_clientlib/src/protocol/example_class.dart
  C:\Users\yaswa\my_appsss\my_appsss_clientlib/src/protocol/protocol.dart
  C:\Users\yaswa\my_appsss\my_appsss_clientpubspec.yaml
  C:\Users\yaswa\my_appsss\my_appsss_clientREADME.md
  C:\Users\yaswa\my_appsss\my_appsss_flutteranalysis_options.yaml
  C:\Users\yaswa\my_appsss\my_appsss_fluttergitignore
  C:\Users\yaswa\my_appsss\my_appsss_flutterlib/main.dart
  C:\Users\yaswa\my_appsss\my_appsss_flutterpubspec.yaml
  C:\Users\yaswa\my_appsss\my_appsss_flutterREADME.md
  C:\Users\yaswa\my_appsss\my_appsss_fluttertest/widget_test.dart
Running `dart pub get` in C:\Users\yaswa\my_appsss\my_appsss_server
Resolving dependencies...
+ args 2.3.0
+ async 2.9.0
+ buffer 1.1.1
+ charcode 1.3.1
+ collection 1.16.0
+ crypto 3.0.1
+ executor 2.2.2
+ http 0.13.4
+ http_parser 4.0.0
+ lints 1.0.1 (2.0.0 available)
+ meta 1.7.0
+ path 1.8.1
+ pedantic 1.11.1
+ postgres 2.4.3
+ redis 2.2.0 (3.0.0 available)
+ retry 3.1.0
+ sasl_scram 0.1.0
+ saslprep 1.0.2
+ serverpod 0.9.6
+ serverpod_client 0.9.6
+ serverpod_postgres_pool 2.1.1
+ serverpod_serialization 0.9.6
+ serverpod_service_client 0.9.6
+ serverpod_shared 0.9.6
+ source_span 1.8.2
+ stack_trace 1.10.0
+ stream_channel 2.1.0
+ string_scanner 1.1.0
+ term_glyph 1.2.0
+ typed_data 1.3.0
+ unorm_dart 0.2.0
+ vm_service 8.2.2
+ web_socket_channel 2.1.0
+ yaml 3.1.0
Downloading serverpod 0.9.6...
Changed 34 dependencies!

Running `dart pub get` in C:\Users\yaswa\my_appsss\my_appsss_client
Resolving dependencies...
+ async 2.9.0
+ charcode 1.3.1
+ collection 1.16.0
+ crypto 3.0.1
+ http 0.13.4
+ http_parser 4.0.0
+ meta 1.7.0
+ path 1.8.1
+ serverpod_client 0.9.6
+ serverpod_serialization 0.9.6
+ source_span 1.8.2
+ stream_channel 2.1.0
+ string_scanner 1.1.0
+ term_glyph 1.2.0
+ typed_data 1.3.0
+ web_socket_channel 2.1.0
+ yaml 3.1.0
Changed 17 dependencies!

Running `flutter create` in C:\Users\yaswa\my_appsss\my_appsss_flutter
Unhandled exception:
ProcessException: The system cannot find the file specified.

  Command: flutter create .
#0      _ProcessImpl._start (dart:io-patch/process_patch.dart:401:33)
#1      Process.start (dart:io-patch/process_patch.dart:38:20)
#2      _runNonInteractiveProcess (dart:io-patch/process_patch.dart:578:18)
#3      Process.run (dart:io-patch/process_patch.dart:49:12)
#4      CommandLineTools.flutterCreate (file:///C:/Users/yaswa/AppData/Local/Pub/Cache/hosted/pub.dartlang.org/serverpod_cli-0.9.6/bin/create/command_line_tools.dart:16:32)
#5      performCreate (file:///C:/Users/yaswa/AppData/Local/Pub/Cache/hosted/pub.dartlang.org/serverpod_cli-0.9.6/bin/create/create.dart:212:28)
<asynchronous suspension>
#6      main (file:///C:/Users/yaswa/AppData/Local/Pub/Cache/hosted/pub.dartlang.org/serverpod_cli-0.9.6/bin/serverpod_cli.dart:218:9)
<asynchronous suspension>
WARNING! Windows is not officially supported yet. Things may or may not work as expected.

WARNING! skipping check to see if "dart" exists.
WARNING! skipping check to see if "flutter" exists.
WARNING! skipping check to see if "docker" exists.
Project my_appsss already exists.
```

**Comment by yahu1031** [comment]
I tested the last version. for some reason `flutter *` command doesn't work as expected. not sure why. Need to investigate it more. I think I know the alternate approach. I can try but gimme some time. Busy with office work 😅

**Comment by anandsubbu007** [comment]
https://github.com/serverpod/serverpod/issues/58#issuecomment-1110041730

**Comment by yahu1031** [comment]
This was fixed in favor of #134
